kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/RecordEncryptionST.java

```diff
@@ -115,24 +128,126 @@ void produceAndConsumeMessage(String namespace, TestKmsFacade<?, ?, ?> testKmsFa
+        assumeThat(testKmsFacade.getKmsServiceClass().getSimpleName().toLowerCase().contains("vault")).isTrue();
+        testKekManager = testKmsFacade.getTestKekManager();
+        testKekManager.generateKek("KEK_" + topicName);
+        int numberOfMessages = 1;
+        ExperimentalKmsConfig experimentalKmsConfig = new ExperimentalKmsConfig(1, 1, 1, 1);
+
+        // start Kroxylicious
+        LOGGER.info("Given Kroxylicious in {} namespace with {} replicas", namespace, 1);
+        Kroxylicious kroxylicious = new Kroxylicious(namespace);
+        kroxylicious.deployPortPerBrokerPlainWithRecordEncryptionFilter(clusterName, 1, testKmsFacade, experimentalKmsConfig);
+        bootstrap = kroxylicious.getBootstrap();
+
+        LOGGER.info("And a kafka Topic named {}", topicName);
+        KafkaSteps.createTopic(namespace, topicName, bootstrap, 1, 2);
+
+        LOGGER.info("When {} messages '{}' are sent to the topic '{}'", numberOfMessages, MESSAGE, topicName);
+        KroxyliciousSteps.produceMessages(namespace, topicName, bootstrap, MESSAGE, numberOfMessages);
+
+        LOGGER.info("Then the messages are consumed");
+        List<ConsumerRecord> resultEncrypted = KroxyliciousSteps.consumeMessageFromKafkaCluster(namespace, topicName, clusterName,
+                Constants.KAFKA_DEFAULT_NAMESPACE, numberOfMessages, Duration.ofMinutes(2));
+        LOGGER.info("Received: {}", resultEncrypted);
+
+        assertKekVersionWithinParcel(resultEncrypted, "v1", testKekManager);
+
+        LOGGER.info("When KEK is rotated");
+        testKekManager.rotateKek("KEK_" + topicName);
+
+        LOGGER.info("And {} messages '{}' are sent to the topic '{}'", numberOfMessages, MESSAGE, topicName);
+        KroxyliciousSteps.produceMessages(namespace, topicName, bootstrap, MESSAGE, numberOfMessages);
+
+        LOGGER.info("Then the messages are consumed");
+        List<ConsumerRecord> resultEncryptedRotatedKek = KroxyliciousSteps.consumeMessageFromKafkaCluster(namespace, topicName, clusterName,
+                Constants.KAFKA_DEFAULT_NAMESPACE, numberOfMessages, Duration.ofMinutes(2));
+        LOGGER.info("Received: {}", resultEncryptedRotatedKek);
+
+        assertKekVersionWithinParcel(resultEncryptedRotatedKek.stream().filter(r -> !r.getValue().contains("v1")).toList(),
+                "v2", testKekManager);
+    }
+
+    private void assertKekVersionWithinParcel(List<ConsumerRecord> consumerRecords, String expectedValue, TestKekManager testKekManager) {
+        assertThat(consumerRecords.size())
+                .withFailMessage("expected messages not received! Consumer records is empty")
+                .isNotZero();
+
+        assertThat(testKekManager.getClass().getSimpleName().toLowerCase())
+                .withFailMessage("Another KMS different from Vault is not currently supported!")
+                .contains("vault");
+
+        if (testKekManager.getClass().getSimpleName().toLowerCase().contains("vault")) {
```

Comment:
This has been removed. We don't need to check that is actually an instance, I wanted to be sure that this is only run when we are using vault, but this is a duplicated check because we are already assuming we are in a vault kms.